### PR TITLE
BS-14269 Undeploy archived process instances before Preview

### PR DESCRIPTION
**Pull request by rbioteau**
Reuse undeploy logic from run operation.

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.engine/src/org/bonitasoft/studio/engine/operation/DeployProcessOperation.java [hunk 1]
```diff
@@ -273,84 +273,81 @@ protected IStatus handleProcessEnablementException(final AbstractProcess process
...
-        }
-    }
+    //    protected void undeployProcess(final AbstractProcess process, final IProgressMonitor monitor) throws Exception {
+    //        final APISession session = BOSEngineManager.getInstance().createSession(process, configurationId, monitor);
+    //        try {
+    //            final ProcessAPI processApi = BOSEngineManager.getInstance().getProcessAPI(session);
+    //            final long nbDeployedProcesses = processApi.getNumberOfProcessDeploymentInfos();
+    //            if (nbDeployedProcesses > 0) {
+    //                final long processDefinitionId = processApi.getProcessDefinitionId(process.getName(), process.getVersion());
+    //                disableProcessDefinition(process, processApi, processDefinitionId, monitor);
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.engine/src/org/bonitasoft/studio/engine/operation/DeployProcessOperation.java]: commented code to remove

File: bundles/plugins/org.bonitasoft.studio.engine/src/org/bonitasoft/studio/engine/operation/UndeployProcessOperation.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (C) 2009-2011 BonitaSoft S.A.
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.engine/src/org/bonitasoft/studio/engine/operation/UndeployProcessOperation.java]: copyright to update,  isn't it?

File: bundles/plugins/org.bonitasoft.studio.form.preview/src/org/bonitasoft/studio/form/preview/FormPreviewOperation.java [hunk 3]
```diff
@@ -268,34 +264,36 @@ private void initializeLookNFeel(AbstractProcess proc, final Resource resource)
...
+    //
+    //    protected void undeployDeployedProcess(final AbstractProcess process, final ProcessAPI processApi) throws InvalidSessionException,
+    //            ProcessDefinitionNotFoundException, IllegalProcessStateException, DeletionException {
+    //        final long nbDeployedProcesses = processApi.getNumberOfProcessDeploymentInfos();
+    //        if (nbDeployedProcesses > 0) {
+    //            if (lastProcessDeployed == null) {
+    //                lastProcessDeployed = process.getName();
+    //            }
+    //            final List<ProcessDeploymentInfo> processes = processApi.getProcessDeploymentInfos(0, (int) nbDeployedProcesses,
+    //                    ProcessDeploymentInfoCriterion.DEFAULT);
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.form.preview/src/org/bonitasoft/studio/form/preview/FormPreviewOperation.java]: commented code

Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.form.preview/src/org/bonitasoft/studio/form/preview/FormPreviewOperation.java]: :fearful: 
Fixed

Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.form.preview/src/org/bonitasoft/studio/form/preview/FormPreviewOperation.java]: I still see the second commented block code in the PR

**Comment by rbioteau** [comment]
:8ball:

**Pull request merged**